app/presigned_urls.py

```diff
@@ -38,33 +38,40 @@ def make_download_url(full_object_path: str, bucket: str) -> dict:
 def make_presigned_url(event, _context, _kwargs):
     bucket_name = "garden-mlflow-models-dev"
     payload = json.loads(event["body"])
-    s3_path = payload.get("s3_path", None)
-    direction = payload.get("direction", None)
+    batch = payload["batch"]
+    direction = payload["direction"]
 
     if direction not in url_makers:
         return {
             "statusCode": 400,
             "body": json.dumps({"message": f"'direction' must be one of {UPLOAD} or {DOWNLOAD}. Got {direction}"}),
         }
 
-    if s3_path is None or not is_probably_valid_object_path(s3_path):
-        message = "'s3_path' must be formatted like '<email address>/<model short name>/model.zip'. "
-        message += f"Got {s3_path}"
-        return {
-            "statusCode": 400,
-            "body": json.dumps({"message": message}),
+    for i, s3_path in enumerate(batch):
+        if not is_probably_valid_object_path(s3_path):
+            message = "'s3_path' must be formatted like '<email address>/<model short name>/model.zip'. "
+            message += f"Got {s3_path}"
+            batch[i] = {
```

Comment:
We should avoid mutating/overwriting our input array. Like make a separate output array and append to it, and then we don't even have to explicitly keep track of indices